torch/lib/c10d/test/ProcessGroupGlooTest.cpp

```diff
@@ -190,6 +190,20 @@ std::vector<std::vector<at::Tensor>> copyTensors(
   return outputs;
 }
 
+std::vector<std::vector<at::Tensor>> waitWork(
+    std::vector<c10::intrusive_ptr<c10d::ProcessGroup::Work>> works) {
+  std::vector<std::vector<at::Tensor>> outputTensors;
+  for (int i = 0; i < works.size(); ++i) {
+    try {
+      works[i]->wait();
+    } catch (const std::exception& ex) {
+      std::cerr << "Exception received: " << ex.what() << std::endl;
+    }
+    outputTensors.emplace_back(works[i]->result());
```

Comment:
nit: push_back is better here, because `result()` returns `std::vector<at::Tensor>` instead of `std::vector<at::Tensor>&`.

Reply:
thanks!

yes, moving to outputTensors keeps at least ref to the output tensors. I thought C++ RVO figures this out even for std::vector<at::Tensor>.

efficiency should not be a concern here as it is test?

Reply:
Didn't realize it's in the test in my previous quick look. Thx for the udpate!

Reply:
oops, std::move here is not good:

stderr: caffe2/torch/lib/c10d/test/ProcessGroupGlooTest.cpp:202:32: error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
    outputTensors.emplace_back(std::move(works[i]->result()));

https://stackoverflow.com/questions/19267408/why-does-stdmove-prevent-rvo

I think regular RVO should be fine here.